c++/src/kj/filesystem-disk-test.c++

```diff
@@ -945,5 +946,30 @@ KJ_TEST("DiskFile holes") {
 }
 #endif
 
+// Ensure the current path is correctly computed.
+//
+// See issue #1425.
+KJ_TEST("DiskFilesystem::computeCurrentPath") {
+  TempDir tempDir;
+  auto dir = tempDir.get();
+
+  // Paths can be PATH_MAX, but the segments which make up that path typically
+  // can't exceed 255 bytes.
+  auto maxPathSegment = std::string(255, 'a');
+
+  // Create a path which exceeds the 256 byte buffer used in
+  // computeCurrentPath.
+  auto subdir = dir->openSubdir(Path({
+    maxPathSegment,
+    maxPathSegment,
+    "some_path_longer_than_256_bytes"
+  }), WriteMode::CREATE | WriteMode::CREATE_PARENT);
+
+  KJ_SYSCALL(fchdir(KJ_ASSERT_NONNULL(subdir->getFd())));
```

Comment:
You need to arrange to change directories back at the end of the test, otherwise:
* All remaining tests (e.g. unit tests in other source files) will run in the wrong directory, possibly leading to unexpected behavior.
* `TempDir`'s destructor won't be able to delete the temporary directory... I'm kind of surprised this isn't causing a failure, actually.

`fchdir` doesn't exist on Windows, so we definitely at least have to wrap this test in `#if !_WIN32`. That's fine since it's testing for a bug that was Unix-specific anyway.

Reply:
Something like...

```
int origDir = open(cwd, O_RDONLY | O_DIRECTORY);
...
KJ_DEFER(fchdir(origDir));
```

Reply:
Sounds good to me! Thank you for all the help and snippets 😄